File: app/app/src/main/java/com/resourceit/app/dao/LoginDao.java
package com.resourceit.app.dao;

import com.resourceit.app.models.LoginModel;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface LoginDao {

    @Query("SELECT * FROM users")
    List<LoginModel> getAll();

    @Query("SELECT * FROM users WHERE userId = :userId")
    LoginModel findById(int userId);

    @Query("DELETE FROM users")
    void deleteall();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(LoginModel... login);

    @Delete
    void delete(LoginModel login);

}

File: app/src/main/java/br/com/rcp/bank/database/dao/LoginDAO.kt
package br.com.rcp.bank.database.dao

import androidx.room.Dao
import androidx.room.Query
import br.com.rcp.bank.database.dao.base.BaseDAO
import br.com.rcp.bank.database.models.Login

@Dao
abstract class LoginDAO: BaseDAO<Login>() {
	// Only one login is saved
	@Query("select * from login where identifier = 1")
	abstract fun getLogin(): Login?

	@Query("delete from login")
	abstract fun clear()
}